### PR TITLE
Accept screen and tag names in awful.rules

### PR DESCRIPTION
**Pull request by psychon**
This fixes https://github.com/awesomeWM/awesome/issues/799
This should not be merged before https://github.com/awesomeWM/awesome/pull/790 is in, because I don't want to make @Elv13's job harder than it already is. I guess I'll have to rebase this afterwards, but the idea can already be reviewed.

### CODE REVIEW

File: lib/awful/rules.lua [hunk 1]
```diff
@@ -195,23 +212,29 @@ end
...
+            c:tags({ t })
         elseif property == "height" or property == "width" or
                 property == "x" or property == "y" then
             local geo = c:geometry();
             geo[property] = value
             c:geometry(geo);
-        elseif property == "focus" then
+        elseif property == "focus" or property == "switchtotag" then
             -- This will be handled below
             (function() end)() -- I haven't found a nice way to silence luacheck here
```
Comment (blueyed) [inline lib/awful/rules.lua]: Could this be with the last `if`, i.e. `elseif property =~ "focus" and property =~ "swtichtotag"` there?

Comment (psychon) [inline lib/awful/rules.lua]: This is to get rid of the "I haven't found a nice way to silence luacheck here", right?
Yeah, that could work. I'll (hopefully remember to) do a separate PR for that. I guess I would do:
```lua
local ignore = { focus = true, switchtotag = true }
....
elseif not ignore[property] then
    if type(c[property]) == "function" then
        c[property](c, value)
    else
        c[property] = value
    end
end
```

**Comment by Elv13** [comment]
+1

All I would change (and I might do it myself after this is merged) is moving the documentation into a new page instead of Awful rules. This will make it more visible and easier to "wikitize". Something like "Managing clients"

**Comment by psychon** [comment]
Rebased onto master. I addressed @blueyed's comment. Since I honestly don't know how, I didn't do anything about @Elv13's comment.

**Comment by Elv13** [comment]
> Since I honestly don't know how, I didn't do anything about @Elv13's comment.

I have an `awful.rules` PR in the pipeline too, so I will do it. Basically, you add a nn-name.md file in the docs folder, modify config.ld and that's about it. Everything else is just markdown.

**Pull request merged**